internal/pkg/cli/local_run.go

```diff
@@ -106,11 +196,195 @@ func (o *localRunOpts) Execute() error {
+			containerPorts[hostPortStr] = containerportStr
+		}
+	}
+
+	env, err := o.store.GetEnvironment(o.appName, o.envName)
+	if err != nil {
+		return fmt.Errorf("get environment %q configuration: %w", o.envName, err)
+	}
+	o.targetEnv = env
+
+	app, err := o.store.GetApplication(o.appName)
+	if err != nil {
+		return fmt.Errorf("get application %q configuration: %w", o.appName, err)
+	}
+	o.targetApp = app
+
+	envSess, err := o.sessProvider.FromRole(env.ManagerRoleARN, env.Region)
+	if err != nil {
+		return fmt.Errorf("get env session: %w", err)
+	}
+	mft, err := workloadManifest(&workloadManifestInput{
+		name:         o.wkldName,
+		appName:      o.appName,
+		envName:      o.envName,
+		interpolator: o.newInterpolator(o.appName, o.envName),
+		ws:           o.ws,
+		unmarshal:    o.unmarshal,
+		sess:         envSess,
+	})
+	if err != nil {
+		return err
+	}
+	o.appliedDynamicMft = mft
+	o.repository, err = o.configureClients(o)
+	if err != nil {
+		return err
+	}
+
+	err = o.buildContainerImages(o)
+	if err != nil {
+		return fmt.Errorf("building container image: %w", err)
+	}
+
+	containerNames := o.out.ContainerNames
+	imageNames := o.out.ImageNames
+
+	secretsList := make(map[string]string)
+	for _, s := range decryptedSecrets {
+		secretsList[s.Name] = s.Value
+	}
+
+	for i, image := range imageNames {
+		imageInfo := imageInfo{
+			containerName: containerNames[i],
+			imageURI:      image,
+		}
+		o.imageInfoList = append(o.imageInfoList, imageInfo)
+	}
+
+	var sideCarListInfo []imageInfo
+	manifestContent := o.appliedDynamicMft.Manifest()
+	switch t := manifestContent.(type) {
+	case *manifest.ScheduledJob:
+		sideCarListInfo = getBuiltSideCarImages(t.Sidecars)
+	case *manifest.LoadBalancedWebService:
+		sideCarListInfo = getBuiltSideCarImages(t.Sidecars)
+	case *manifest.WorkerService:
+		sideCarListInfo = getBuiltSideCarImages(t.Sidecars)
+	case *manifest.BackendService:
+		sideCarListInfo = getBuiltSideCarImages(t.Sidecars)
+	}
+	o.imageInfoList = append(o.imageInfoList, sideCarListInfo...)
+
+	err = o.runPauseContainer(containerPorts)
+	if err != nil {
+		return err
+	}
+
+	err = o.runContainers(o.imageInfoList, secretsList, envVars)
+	if err != nil {
+		return err
+	}
+
+	return nil
+}
+
+func getBuiltSideCarImages(sidecars map[string]*manifest.SidecarConfig) []imageInfo {
+	//get the image URI for the sidecars which are in a registry
+	sideCarBuiltImages := make(map[string]string)
+	for sideCarName, sidecar := range sidecars {
+		if uri, hasLocation := sidecar.ImageURI(); hasLocation {
+			sideCarBuiltImages[sideCarName] = uri
+		}
+	}
+	var sideCarBuiltImageInfo []imageInfo
+	for sidecarName, image := range sideCarBuiltImages {
+		imageInfo := imageInfo{
+			containerName: sidecarName,
+			imageURI:      image,
+		}
+		sideCarBuiltImageInfo = append(sideCarBuiltImageInfo, imageInfo)
+	}
+	return sideCarBuiltImageInfo
+}
+
+func (o *localRunOpts) runPauseContainer(containerPorts map[string]string) error {
```

Comment:
i think passing a parent context into both this and `runContainers` will be helpful in stopping the running containers!